salon_creator/lib/authentication/sign_in.dart

```diff
@@ -20,6 +23,40 @@ Future<UserCredential> signInWithGoogle() async {
   return await FirebaseAuth.instance.signInWithCredential(credential);
 }
 
+String generateNonce([int length = 32]) {
+  final charset =
+      '0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyz-._';
+  final random = Random.secure();
+  return List.generate(length, (_) => charset[random.nextInt(charset.length)])
+      .join();
+}
+
+String sha256ofString(String input) {
+  final bytes = utf8.encode(input);
+  final digest = sha256.convert(bytes);
+  return digest.toString();
+}
+
+Future<UserCredential> signInWithApple() async {
+  final rawNonce = generateNonce();
+  final nonce = sha256ofString(rawNonce);
+
+  final appleCredential = await SignInWithApple.getAppleIDCredential(
+    scopes: [
+      AppleIDAuthorizationScopes.email,
+      AppleIDAuthorizationScopes.fullName,
+    ],
+    nonce: nonce,
+  );
+
+  final oauthCredential = OAuthProvider("apple.com").credential(
```

Comment:
We keep simple strings (without interpolation) in single quotes!

```suggestion
  final oauthCredential = OAuthProvider('apple.com').credential(
```